syntax = "proto3";

service InviteService {
  rpc get_invites_by_user_id(InvitesRequestByUserId) returns (InvitesResponseByUserId){}
}

message InvitesRequestByUserId {
  string user_id = 1;
}

message Invite {
  string id = 1;
  string event_id = 2;
  string author_id = 3;
  string invitee_id = 4;
  InviteStatus status = 5;
}

message ListOfInvites {
  repeated Invite invites = 1;
}

message InvitesResponseByUserId {
  int64 code = 1;
  optional ListOfInvites invites = 2;
  optional string message = 3;
}

enum InviteStatus {
  PENDING = 0;
  ACCEPTED = 1;
  REJECTED = 2;
}